### PR TITLE
Add "development" docu

### PR DESCRIPTION
**Pull request by vogdb**
@vogdb @tomaszdurka 

- How to install dependencies
- How to run tests
- How to release
 - Travis will release to npm automatically?
 - How to create the tag? Needs to be created *before* the travis build?

### CODE REVIEW

**Comment by vogdb** [comment]
@njam please look

**Comment by njam** [comment]
The reason why I originally opened the issue was because travis did not publish to npm. The reason was that there was no tag on the commit. I guess it's because of `deploy: on: tags: true` in the travis configuration.
What is the recommended way of publishing to npm? Bump the version in code, and…?

@tomaszdurka could you review this as well?

**Comment by vogdb** [comment]
I agree. Can't find the reason why it wasn't published. According to the log nothing critical happened.
```
Installing deploy dependencies

Preparing deploy
NPM version: 2.14.9
Authenticated with email tech@cargomedia.ch
/usr/lib/git-core/git-stash: 186: /usr/lib/git-core/git-stash: cannot create /home/travis/build/cargomedia/cm-janus/.git/logs/refs/stash: Directory nonexistent

Deploying application
NPM API key format changed recently. If your deployment fails, check your API key in ~/.npmrc.
http://docs.travis-ci.com/user/deployment/npm/
~/.npmrc size: 48
--janus@0.4.0
No stash found.
Done. Your build exited with 0.
```

**Comment by njam** [comment]
True, it worked here! https://travis-ci.org/cargomedia/cm-janus/builds/104576336

@tomaszdurka could you review the docu?

File: README.md [hunk 1]
```diff
@@ -45,8 +53,18 @@ jobManager:
...
+## Testing
+cm-janus uses [node-inotify](https://github.com/c4milo/node-inotify) that works only in GNU/Linux. To run tests on any other platform you need to setup a virtual Linux environment. For Vagrant users there is a prepared vagrant file.
+ - So start vagrant `vagrant up`
+ - Tunnel `vagrant ssh`
+ - Go to cm-janus dir `cd ./cm-janus`
+ - Run tests `npm run-script test`
+If you are already in Linux then run only last two steps.
+
+## Publishing
+If for some reason automatic publish didn't happen, you should create a git tag manually and publish it to npm manually `npm publish https://github.com/cargomedia/cm-janus/archive/GitTag.tar.gz`.
```
Comment (tomaszdurka) [inline README.md]: Let's describe normal procedure:
- update package.json with new tag
- release new git tag

If this won't automatically release to npm we should:
- publish to npm manually

**Comment by vogdb** [comment]
@tomaszdurka please review

File: README.md [hunk 2]
```diff
@@ -45,8 +53,24 @@ jobManager:
...
+ - Run tests `npm run-script test`
+If you are already in Linux then run only last two steps.
+
+## Publishing
+ - update package.json with a new version
+ - release a new git tag with the updated package.json
+
+After that the npm release should be done automatically. If it didn't happen then release it manually:
+```
+npm publish https://github.com/cargomedia/cm-janus/archive/GitTagWithUpdatedPackageJson.tar.gz
```
Comment (tomaszdurka) [inline README.md]: Maybe let's rename `GitTagWithUpdatedPackageJson` to `<GitTagWithUpdatedPackageJson>` as I understand this is kinda variable?

Comment (tomaszdurka) [inline README.md]: As first I thought it's the same file every time.

**Comment by vogdb** [comment]
@tomaszdurka please review

**Comment by tomaszdurka** [comment]
Looks good!

**Pull request merged**